test/contlinear.jl

```diff
@@ -726,6 +754,241 @@ function linear7test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+        if MOI.getattribute(m, MOI.ResultCount()) == 1
+            # solver returned an unbounded ray
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.InfeasibilityCertificate
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), [x, y])
+            ray = MOI.getattribute(m, MOI.VariablePrimal(), [x,y])
+            @test ray[1] ≈ ray[2] atol=ε
+
+        else
+            # solver returned nothing
+            @test MOI.getattribute(m, MOI.ResultCount()) == 0
+            @test MOI.cangetattribute(m, MOI.PrimalStatus(1)) == false
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.UnboundedNoResult ||
+                MOI.getattribute(m, MOI.TerminationStatus()) == MOI.InfeasibleOrUnbounded
+        end
+    end
+end
+
+function linear9test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+    @testset "addconstraints" begin
+        #   maximize 1000 x + 350 y
+        #
+        #       s.t.                x >= 30
+        #                           y >= 0
+        #                 x -   1.5 y >= 0
+        #            12   x +   8   y <= 1000
+        #            1000 x + 300   y <= 70000
+        #
+        #   solution: (59.0909, 36.3636)
+        #   objv: 71818.1818
+        @test MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64},
+            [
+                (MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}),
+                (MOI.ScalarAffineFunction{Float64},MOI.LessThan{Float64}),
+                (MOI.SingleVariable,MOI.GreaterThan{Float64})
+            ]
+        )
+
+        m = MOI.SolverInstance(solver)
+        x = MOI.addvariable!(m)
+        y = MOI.addvariable!(m)
+
+        MOI.addconstraints!(m,
+            [MOI.SingleVariable(x), MOI.SingleVariable(y)],
+            [MOI.GreaterThan(30.0), MOI.GreaterThan(0.0)]
+        )
+
+        MOI.addconstraints!(m,
+            [MOI.ScalarAffineFunction([x, y], [1.0, -1.5], 0.0)],
+            [MOI.GreaterThan(0.0)]
+        )
+
+        MOI.addconstraints!(m,
+            [
+                MOI.ScalarAffineFunction([x, y], [12.0, 8.0], 0.0),
+                MOI.ScalarAffineFunction([x, y], [1_000.0, 300.0], 0.0)
+            ],
+            [
+                MOI.LessThan(1_000.0),
+                MOI.LessThan(70_000.0)
+            ]
+        )
+
+        MOI.setobjective!(m, MOI.MaxSense,
+            MOI.ScalarAffineFunction([x, y], [1_000.0, 350.0], 0.0)
+        )
+
+        MOI.optimize!(m)
+
+        @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+        @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+
+        @test MOI.getattribute(m, MOI.ObjectiveValue()) ≈ 79e4/11 atol=ε
+        @test MOI.getattribute(m, MOI.VariablePrimal(), x) ≈ 650/11 atol=ε
+        @test MOI.getattribute(m, MOI.VariablePrimal(), y) ≈ 400/11 atol=ε
```

Comment:
Solvers are able to specify different tolerances... for example https://github.com/JuliaOpt/CPLEX.jl/pull/139/files#diff-fce720c43af3c52c862fd7451c7374b8R30